kentikapi/client.go

```diff
@@ -57,31 +60,100 @@ type Client struct {
 	SyntheticsAdmin grpcsynthetics.SyntheticsAdminServiceClient
 	SyntheticsData  grpcsynthetics.SyntheticsDataServiceClient
 
-	config Config
+	config config
 }
 
-// Config holds configuration of the client.
-type Config struct {
-	// APIURL defaults to "https://api.kentik.com"
+type config struct {
 	APIURL string
 
 	AuthEmail string
 	AuthToken string
 	RetryCfg  RetryConfig
 
-	// LogPayloads enables logging of request and response payloads to Cloud Export and Synthetics APIs.
 	LogPayloads bool
-	// Timeout specifies a limit of a total time of a single client call, including redirects and retries.
-	// A Timeout of zero means no timeout.
-	// Default: 100 seconds.
-	Timeout *time.Duration
+
+	Timeout time.Duration
 }
 
 type RetryConfig = httputil.RetryConfig
 
+// ClientOption defines an option for a Client.
+type ClientOption func(*config)
+
+// WithAPIURL defaults to "https://api.kentik.com".
+func WithAPIURL(apiURL string) ClientOption {
+	return func(c *config) {
+		c.APIURL = apiURL
+	}
+}
+
+// WithCredentials specifies authorization email and token for a Client.
+func WithCredentials(authEmail, authToken string) ClientOption {
+	return func(c *config) {
+		c.AuthEmail = authEmail
+		c.AuthToken = authToken
+	}
+}
+
+// WithTimeout specifies a limit of a total time of a single client call, including redirects and retries.
+// A Timeout of zero means no timeout.
+// Default: 100 seconds.
+func WithTimeout(timeout time.Duration) ClientOption {
+	return func(c *config) {
+		if pointer.ToDurationOrNil(timeout) != nil {
```

Comment:
```suggestion
		c.Timeout = &timeout
```
This is a bug, because that way user cannot set 0 timeout (no timeout).

Reply:
Done. However underlying code does not handle 0 as no timeout so I added suggestion to remove that docstring.